### ISSUE TITLE
'GraphQLScalarType' is declared but its value is never read when "federation" is true

### ISSUE THREAD

**Issue by ForbesLindesay** [dotansimha/graphql-code-generator#2541]
**Describe the bug**
 
When I generate types for a schema with no custom scalars, I get the TypeScript errors:

```
src/graphql/__generated__/types.ts:5:30 - error TS6133: 'GraphQLScalarType' is declared but its value is never read.

5 import { GraphQLResolveInfo, GraphQLScalarType, GraphQLScalarTypeConfig } from 'graphql';
                               ~~~~~~~~~~~~~~~~~

src/graphql/__generated__/types.ts:5:49 - error TS6133: 'GraphQLScalarTypeConfig' is declared but its value is never read.

5 import { GraphQLResolveInfo, GraphQLScalarType, GraphQLScalarTypeConfig } from 'graphql';
```

**To Reproduce**

1. https://codesandbox.io/embed/graphql-codegen-issue-template-uefpt
2. run yarn generate
3. run yarn typecheck

1. My GraphQL schema:

```graphql
type Query {
    user(id: ID!): User!
}

type User {
    id: ID!
    username: String!
    email: String!
}
```

3. My `codegen.yml` config file:

```yml
schema: schema.graphql
generates:
  types.ts:
    plugins:
      - typescript
      - typescript-resolvers
    config:
      federation: true
```

4. tsconfig.json

```json
{
  "compilerOptions": {
    "noUnusedLocals": true
  },
  "include": [
    "types.ts"
  ]
}
```

**Comment by kamilkisiela** [comment]
#2564

**Comment by dotansimha** [comment]
Fixed in 1.8.0 🚀
